src/python/pants/task/task.py

```diff
@@ -194,14 +194,35 @@ def get_options(self):
     return self.context.options.for_scope(self.options_scope)
 
   def get_passthru_args(self):
-    """
+    """Returns the passthru args for this task, if it supports them.
+
     :API: public
     """
     if not self.supports_passthru_args():
       raise TaskError('{0} Does not support passthru args.'.format(self.stable_name()))
     else:
       return self.context.options.passthru_args_for_scope(self.options_scope)
 
+  # NOTE: This method was introduced in 2018, so at the time of writing few tasks consult it.
+  # Instead, they query self.context.targets directly.
+  # TODO: Fix up existing targets to consult this method, for uniformity.
+  def get_targets(self, predicate=None):
+    """Returns the candidate targets this task should act on.
+
+    By default returns the entire transitive dependency closure of the target roots
+    specified on the command line.  Subclasses can override to provide more selective
```

Comment:
Should probably explicitly reference the mixins and discourage custom overrides.

Reply:
I can mention them, but I think it's OK to trust the programmer to know they have to get this right. "It's a language feature - use with awareness..." :)